### PR TITLE
[WIP] CORS dispatching tests

### PR DESCRIPTION
**Pull request by JamesNK**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs [hunk 1]
```diff
@@ -342,7 +344,17 @@ private static EndpointMetadataCollection BuildEndpointMetadata(ActionDescriptor
                 {
                     if (actionConstraint is HttpMethodActionConstraint httpMethodActionConstraint)
                     {
-                        metadata.Add(new HttpMethodEndpointConstraint(httpMethodActionConstraint.HttpMethods));
+                        var httpEndpointConstraint = new HttpMethodEndpointConstraint(httpMethodActionConstraint.HttpMethods);
+
+                        // yolo
+                        if (httpMethodActionConstraint.GetType().Name == "CorsHttpMethodActionConstraint")
```
Comment (JamesNK) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs]: Sorry not sorry.

Comment (JamesNK) [reply src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs]: But seriously CORS is off in its own project so maybe we need to think about extensiblity model for EndpointDataSource and handling situations like this.

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs [hunk 2]
```diff
@@ -443,4 +455,51 @@ public RouteNameMetadata(string routeName)
             public string Name { get; }
         }
     }
+
+    public class CorsHttpMethodEndpointConstraint : HttpMethodEndpointConstraint, IActionConstraintMetadata
```
Comment (JamesNK) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs]: Temporarily placed here.

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/CorsTests.cs [hunk 3]
```diff
@@ -1,354 +1,13 @@
...
-using System.Linq;
-using System.Net;
-using System.Net.Http;
-using System.Threading.Tasks;
-using Microsoft.AspNetCore.Cors.Infrastructure;
-using Xunit;
-
 namespace Microsoft.AspNetCore.Mvc.FunctionalTests
 {
-    public class CorsTests : IClassFixture<MvcTestFixture<CorsWebSite.Startup>>
```
Comment (JamesNK) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/CorsTests.cs]: Everything past here is copy/paste

**Comment by JamesNK** [comment]
Be my guest

**Comment by rynowak** [comment]
guest'd

**Comment by JamesNK** [comment]
https://github.com/aspnet/Mvc/pull/8158

**Pull request closed without merging**